packages/service-content/src/lib/resources/import/categories/newsletters.ts

```diff
@@ -0,0 +1,127 @@
+import { validate as uuidValidate } from 'uuid';
+
+import { firstRow } from '@blms/database';
+import type { Resource } from '@blms/types';
+
+import type { Dependencies } from '../../../dependencies.js';
+import { separateContentFiles, yamlToObject } from '../../../utils.js';
+import type { BaseResource, ChangedResource } from '../index.js';
+import { createProcessMainFile } from '../main.js';
+
+interface NewsletterMain {
+  id: string;
+  level: string;
+  author: string;
+  title: string;
+  link?: Array<{ website?: string }>;
+  publication_date: string;
+  tags?: string[];
+  contributors?: string[];
+  language: string;
+}
+
+interface NewsletterLocal extends BaseResource {
+  description: string;
+}
+
+export const createProcessChangedNewsletter = (
+  { postgres }: Dependencies,
+  errors: string[],
+) => {
+  return async (resource: ChangedResource) => {
+    return postgres
+      .begin(async (transaction) => {
+        const { main, files } = separateContentFiles(
+          resource,
+          'newsletter.yml',
+        );
+        if (!main) return;
+
+        try {
+          const processMainFile = createProcessMainFile(transaction);
+          await processMainFile(resource, main);
+        } catch (error) {
+          errors.push(
+            `Error processing file(newsletters) ${resource?.path} (${resource.fullPath}): ${error}`,
+          );
+          return;
+        }
+
+        const id = await transaction<Resource[]>`
+          SELECT id FROM content.resources WHERE path = ${resource.path}
+        `
+          .then(firstRow)
+          .then((row) => row?.id);
+
+        if (!id) {
+          throw new Error(`Resource not found for path ${resource.path}`);
+        }
+
+        try {
+          const parsed = yamlToObject<NewsletterMain>(main.data);
+
+          if (!uuidValidate(parsed.id)) {
+            throw new Error(`Invalid UUID format for parsed.id: ${parsed.id}`);
+          }
+          const websiteUrl = parsed.link?.[0]?.website;
+          await transaction`
+            INSERT INTO content.newsletters (
+              resource_id, id, author, level, website_url, publication_date, title, tags, contributors, language
+            )
+            VALUES (
+              ${id},
+              ${parsed.id}::uuid,
+              ${parsed.author},
+              ${parsed.level},
+              ${websiteUrl},
+              ${parsed.publication_date},
+              ${parsed.title},
+              ${parsed.tags || []},
+              ${parsed.contributors || []},
+              ${parsed.language}
+            )
+            ON CONFLICT (resource_id) DO UPDATE SET
```

Comment:
Check general comment concerning uuid vs resource_id -> shouldn't we start using the uuid of the ressource instead ? @trigger67